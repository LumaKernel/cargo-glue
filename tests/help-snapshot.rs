use insta::assert_snapshot;
use std::{process::Output, str};

#[test]
fn helps() -> anyhow::Result<()> {
    assert_snapshot!("short", run("-h")?);
    assert_snapshot!("long", run("--help")?);
    Ok(())
}

fn run(flag: &str) -> anyhow::Result<String> {
    let assert = assert_cmd::Command::cargo_bin("cargo-glue")?
        .args(["glue", flag])
        .assert()
        .success();
    let Output { stdout, .. } = assert.get_output();
    Ok(str::from_utf8(stdout)?.replacen(env!("CARGO_PKG_VERSION"), "<version>", 1))
}
