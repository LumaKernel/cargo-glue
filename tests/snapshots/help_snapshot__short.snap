---
source: tests/help-snapshot.rs
expression: "run(\"-h\")?"
---
cargo-equip <version>
Ryo Yamashita <qryxip@gmail.com>

A Cargo subcommand to bundle your code into one `.rs` file for competitive programming.

Use -h for short descriptions and --help for more detials.

USAGE:
    cargo equip [OPTIONS]
    cargo equip [OPTIONS] --lib
    cargo equip [OPTIONS] --bin <NAME>
    cargo equip [OPTIONS] --example <NAME>
    cargo equip [OPTIONS] --src <PATH>

OPTIONS:
        --src <PATH>                                  Bundle the lib/bin/example target and its dependencies
        --lib                                         Bundle the library and its dependencies
        --bin <NAME>                                  Bundle the binary and its dependencies
        --example <NAME>                              Bundle the binary example and its dependencies
        --manifest-path <PATH>                        Path to Cargo.toml
        --exclude <SPEC>...                           Exclude library crates from bundling
        --exclude-atcoder-crates                      Alias for `--exclude {crates available on AtCoder}`
        --exclude-atcoder-202301-crates               Alias for `--exclude {crates available on AtCoder 202301}`
        --exclude-atcoder-202308-crates               Alias for `--exclude {crates available on AtCoder 202308}`
        --exclude-codingame-crates                    Alias for `--exclude {crates available on CodinGame}`
        --mine <DOMAIN_AND_USERNAME>...               Do not include license and copyright notices for the users
        --toolchain-for-udeps <TOOLCHAIN>             `nightly` toolchain for `cargo-udeps` [default: nightly]
        --toolchain-for-proc-macro-srv <TOOLCHAIN>    Toolchain for expanding procedural macros
        --mod-path <MODULE_PATH>                      Expand the libraries to the module [default: crate::__cargo_equip]
        --remove <REMOVE>...                          Remove some part [possible values: docs, comments]
        --minify <MINIFY>
            Minify part of the output before emitting [default: none]  [possible values: none, libs, all]

        --no-resolve-cfgs                             Do not resolve `cfg(..)`s
        --no-rustfmt                                  Do not format the output before emitting
        --no-check                                    Do not check the output before emitting
    -o, --output <PATH>                               Write to the file instead of STDOUT
    -h, --help                                        Prints help information
    -V, --version                                     Prints version information
